scipy/signal/windows/windows.py

```diff
@@ -1620,6 +1620,76 @@ def exponential(M, center=None, tau=1., sym=True):
     return _truncate(w, needs_trunc)
 
 
+def taylor(M, nbar=4, level=-30, sym=True):
+    """
+    Return the Taylor window.
+
+    The Taylor window allows for a selectable sidelobe suppression with a
+    minimum broadening. This window is commonly used in radar processing [1]_.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an
+        empty array is returned.
+    nbar : int
+        Number of nearly constant level sidelobes adjacent to the mainlobe
+    level : float
+        Desired peak sidelobe level in decibels (db) relative to the mainlobe
+    sym : bool, optional
+        When True (default), generates a symmetric window, for use in filter
+        design.
+        When False, generates a periodic window, for use in spectral analysis.
+
+    Returns
+    -------
+    out : array
+        The window, with the center value normalized to one (the value
+        one appears only if the number of samples is odd).
+
+    See Also
+    --------
+    kaiser, bartlett, blackman, hamming, hanning
+
+    References
+    -----
+    .. [1] W. Carrara, R. Goodman, and R. Majewski, "Spotlight Synthetic
+            Aperture Radar: Signal Processing Algorithms" Pages 512-513,
+            July 1995.
+    .. [2] Armin Doerry, "Catalog of Window Taper Functions for
+            Sidelobe Control", 2017.
+            https://www.researchgate.net/profile/Armin_Doerry/publication/316281181_Catalog_of_Window_Taper_Functions_for_Sidelobe_Control/links/58f92cb2a6fdccb121c9d54d/Catalog-of-Window-Taper-Functions-for-Sidelobe-Control.pdf
+    """
```

Comment:
Should get CIs to ignore the line length error above
```suggestion
    """  # noqa: E501
```

Reply:
Doesn't it normally ignore long URLs on a line by themselves?  Is it from being indented one space too many?

```
    References
    ----------
    .. [1] DSPRelated, "Generalized Hamming Window Family",
           https://www.dsprelated.com/freebooks/sasp/Generalized_Hamming_Window_Family.html
    .. [2] Wikipedia, "Window function",
           https://en.wikipedia.org/wiki/Window_function
    .. [3] Riccardo Piantanida ESA, "Sentinel-1 Level 1 Detailed Algorithm
           Definition",
           https://sentinel.esa.int/documents/247904/1877131/Sentinel-1-Level-1-Detailed-Algorithm-Definition
    .. [4] Matthieu Bourbigot ESA, "Sentinel-1 Product Definition",
           https://sentinel.esa.int/documents/247904/1877131/Sentinel-1-Product-Definition
```
vs
```
    References
    -----
    .. [1] W. Carrara, R. Goodman, and R. Majewski, "Spotlight Synthetic
            Aperture Radar: Signal Processing Algorithms" Pages 512-513,
            July 1995.
    .. [2] Armin Doerry, "Catalog of Window Taper Functions for
            Sidelobe Control", 2017.
            https://www.researchgate.net/profile/Armin_Doerry/publication/316281181_Catalog_of_Window_Taper_Functions_for_Sidelobe_Control/links/58f92cb2a6fdccb121c9d54d/Catalog-of-Window-Taper-Functions-for-Sidelobe-Control.pdf
```

Reply:
It looks like these weren't really the issue (for the linter that is, but I have fixed them to be proper RST for doc generation). The linter long line options were from a different doc string plus this line:
```
    - `~scipy.signal.windows.taylor` (needs number of constant sidelobes, sidelobe level)
```

https://github.com/scipy/scipy/pull/12777/files#diff-51249363916e71b8020bdfe9a29c697aR2047

I am not sure how to break up this line or if I should add the noqa here. I assume we'll just add "noqa" to the end of this docstring too?

Reply:
Yes end of the docstring is fine, or you can break up the RST such that the first chars line up like:
```
    - `~scipy.signal.windows.taylor` (needs number of constant
      sidelobes, sidelobe level)
```
Up to you

Reply:
See latest commit for length fixes.